addons/calendar/models/calendar_event.py

```diff
@@ -669,8 +678,15 @@ def action_join_meeting(self, partner_id):
             self.write({'partner_ids': [(4, partner.id)]})
 
     def action_mass_deletion(self, recurrence_update_setting):
+        """
+        Action used to delete all events (and the recurrence itself) or all events from self from a recurrence,
+        according to 'recurrence_update_setting' parameter.
+        Note that deleting all events and futures from the first one means deleting all events and the recurrence
+        itself.
+        """
         self.ensure_one()
-        if recurrence_update_setting == 'all_events':
+
+        if recurrence_update_setting == 'all_events' or self == self.recurrence_id.base_event_id:
```

Comment:
Hum, I'm not sure to understand what you mean.
If the user deletes the base event without selecting "and future events" or "all events", it will delete only this event and the next event will become the base event.
But, if the user deletes the base event and select "and future events", instead of deleting all events without the `calendar.recurrence` record itself, it will also delete the `calendar.recurrence` record :-)

Reply:
Indeed that's clever. Sorry for the noise.

Reply:
no problem, it's always good to challenge modifications ;-)